### PR TITLE
API1000 support for Volume template

### PR DESCRIPTION
**Pull request by madhav-bharadwaj**
### Description
API1000 Support for Storage volume templates

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass for go 1.11 + gofmt checks.
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: endpoints-support.md [hunk 1]
```diff
@@ -182,8 +182,14 @@
...
 |<sub>/rest/switch-types/{id}</sub>                                                       | GET      | :white_check_mark:   | :white_check_mark:   | :heavy_multiplication_x:   |
 |     **Storage Volume Attachments**                                                                                                                                   |
-|<sub>/rest/storage-volume-attachments</sub>                                              | GET      | :heavy_multiplication_x:  | :heavy_multiplication_x: | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | 
-|<sub>/rest/storage-volume-attachments/{id}</sub>                                         | GET      | :heavy_multiplication_x:  | :heavy_multiplication_x: | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
+|<sub>/rest/storage-volume-attachments</sub>                                              | GET      | :heavy_multiplication_x:  | :heavy_multiplication_x: | :heavy_multiplication_x:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
```
Comment (AnushaY1916) [inline endpoints-support.md]: We are following alphabetical order for the resources, Can you please place the Switch Types below storage volume templates.

File: endpoints-support.md [hunk 2]
```diff
@@ -182,8 +182,14 @@
...
+|<sub>/rest/storage-volume-attachments/{id}</sub>                                         | GET      | :heavy_multiplication_x:  | :heavy_multiplication_x: | :heavy_multiplication_x:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
+|     **Storage Volume Templates**                                                                                                                                   |
+|<sub>/rest/storage-volume-templates</sub>                                              | GET      | :heavy_multiplication_x:  | :heavy_multiplication_x: | :heavy_multiplication_x:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | 
+|<sub>/rest/storage-volume-templates/{id}</sub>                                         | GET      | :heavy_multiplication_x:  | :heavy_multiplication_x: | :heavy_multiplication_x:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
+|<sub>/rest/storage-volume-templates</sub>                                                         | POST     | :heavy_multiplication_x:   | :heavy_multiplication_x:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
```
Comment (AnushaY1916) [inline endpoints-support.md]: Only POST is supported in v500. Did you test only POST or is it by mistake?

File: examples/storage_volume_template.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+	err := ovc.CreateStorageVolumeTemplate(storageVolumeTemplate)
+	if err != nil {
+		fmt.Println("Could not create the volume Template", err)
+	} else {
+		fmt.Println("Volume Template created successfully", storageVolumeTemplate.Name)
+	}
+
+	// Get the volume template by name
+	update_vol_template, _ := ovc.GetStorageVolumeTemplateByName(name_to_create)
```
Comment (AnushaY1916) [inline examples/storage_volume_template.go]: Error handling is needed

File: ov/storage_volume_template.go [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+	Status                      string             `json:"status,omitempty"`
+	Type                        string             `json:"type,omitempty"`
+	URI                         utils.Nstring      `json:"uri,omitempty"`
+	InitialScopeUris            utils.Nstring      `json:"initialScopeUris,omitempty"`
+	RootTemplateUri             utils.Nstring      `json:"rootTemplateUri,omitempty"`
+	StoragePoolUri              utils.Nstring      `json:"storagePoolUri,omitempty"`
+	Version                     string             `json:"version,omitempty"`
+	Uuid                        string             `json:"uuid,omitempty"`
+	Family                      string             `json:"family,omitempty"`
+	TemplateProperties          TemplateProperties `json:"properties,omitempty"`
```
Comment (AnushaY1916) [inline ov/storage_volume_template.go]: Can you make it a pointer?

File: endpoints-support.md (hunk 1 above)
Comment (madhav-bharadwaj) [reply endpoints-support.md]: Done.

File: examples/storage_volume_template.go (hunk 3 above)
Comment (madhav-bharadwaj) [reply examples/storage_volume_template.go]: Have added it

File: endpoints-support.md (hunk 2 above)
Comment (madhav-bharadwaj) [reply endpoints-support.md]: Corrected it

File: ov/storage_volume_template.go (hunk 4 above)
Comment (madhav-bharadwaj) [reply ov/storage_volume_template.go]: Done.

File: ov/storage_volume_template.go [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+	Description                 utils.Nstring       `json:"description,omitempty"`
+	ETAG                        string              `json:"eTag,omitempty"`
+	IsRoot                      bool                `json:"isRoot,omitempty"`
+	ScopesURI                   utils.Nstring       `json:"scopesUri,omitempty"`
+	Name                        string              `json:"name,omitempty"`
+	State                       string              `json:"state,omitempty"`
+	Status                      string              `json:"status,omitempty"`
+	Type                        string              `json:"type,omitempty"`
+	URI                         utils.Nstring       `json:"uri,omitempty"`
+	InitialScopeUris            utils.Nstring       `json:"initialScopeUris,omitempty"`
```
Comment (AnushaY1916) [inline ov/storage_volume_template.go]: InitialScopeUris should be a slice.

File: ov/storage_volume_template.go [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+	VolumetSet                    *TemplatePropertyDatatypeStruct     `json:"volumeSet,omitempty"`
+	Description                   *TemplatePropertyDatatypeStruct     `json:"description,omitempty"`
+	IsAdaptiveOptimizationEnabled *TemplatePropertyDatatypeStructBool `json:"isAdaptiveOptimizationEnabled,omitempty"`
+	IsCompressed                  *TemplatePropertyDatatypeStruct     `json:"isCompressed,omitempty"`
+	DataProtectionLevel           *TemplatePropertyDatatypeStruct     `json:"dataProtectionLevel,omitempty"`
+	IsShareable                   *TemplatePropertyDatatypeStructBool `json:"isShareable,omitempty"`
+}
+
+type TemplatePropertyDatatypeStruct struct {
+	Meta        Meta          `json:"meta,omitempty"`
```
Comment (AnushaY1916) [inline ov/storage_volume_template.go]: Can these Meta be pointer in all the structs used

Comment (madhav-bharadwaj) [reply ov/storage_volume_template.go]: Changed it.

**Pull request merged**